pkg/liqoctl/info/peer/auth.go

```diff
@@ -0,0 +1,217 @@
+// limitations under the License.
+//
+
+package peer
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	authv1beta1 "github.com/liqotech/liqo/apis/authentication/v1beta1"
+	liqov1beta1 "github.com/liqotech/liqo/apis/core/v1beta1"
+	"github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/liqoctl/info"
+	"github.com/liqotech/liqo/pkg/liqoctl/info/common"
+	"github.com/liqotech/liqo/pkg/liqoctl/output"
+	"github.com/liqotech/liqo/pkg/utils/getters"
+)
+
+// ResourceSliceAction tells whether the cluster is consuming or providing the resources.
+type ResourceSliceAction string
+
+const (
+	// ConsumingAction tells that the cluster is requesting and consuming the resources.
+	ConsumingAction ResourceSliceAction = "Consuming"
+	// ProvidingAction tells that the cluster is providing the resources.
+	ProvidingAction ResourceSliceAction = "Providing"
+)
+
+// ResourceSliceStatus contains info about a ResourceSliceStatus CR.
+type ResourceSliceStatus struct {
+	Name      string              `json:"name"`
+	Action    ResourceSliceAction `json:"action"`
+	Accepted  bool                `json:"accepted"`
+	Alerts    []string            `json:"alerts,omitempty"`
+	Resources corev1.ResourceList `json:"resources"`
+}
+
+// Auth contains some info about the current status of the authentication module.
+type Auth struct {
+	Status         common.ModuleStatus `json:"status"`
+	Alerts         []string            `json:"alerts,omitempty"`
+	APIServerAddr  string
+	ResourceSlices []ResourceSliceStatus `json:"resourceSlices"`
+}
+
+// AuthChecker collects some info about the current status of the authentication module.
+type AuthChecker struct {
+	info.CheckerCommon
+
+	// In this case data is a mapping between ClusterID and peer Authentication info
+	data map[liqov1beta1.ClusterID]Auth
+}
+
+// Collect some info about the current status of the authentication module.
+func (ac *AuthChecker) Collect(ctx context.Context, options info.Options) {
+	ac.data = map[liqov1beta1.ClusterID]Auth{}
+	for clusterID := range options.ClustersInfo {
+		authStatus := Auth{}
+		ac.collectStatusInfo(clusterID, options.ClustersInfo, &authStatus)
+
+		if options.ClustersInfo[clusterID].Status.Role == liqov1beta1.ProviderRole {
+			if err := ac.collectAPIAddress(ctx, options.CRClient, clusterID, &authStatus); err != nil {
+				ac.AddCollectionError(fmt.Errorf("unable to get API server address of cluster %q: %w", clusterID, err))
+			}
+		}
+
+		labelFilter, _ := labels.NewRequirement(consts.RemoteClusterID, selection.Equals, []string{string(clusterID)})
+		// Get the ResourceSlices related to the given remote clusterID
+		var resourceSliceList authv1beta1.ResourceSliceList
+		if err := options.CRClient.List(ctx, &resourceSliceList, &client.ListOptions{
+			LabelSelector: labels.NewSelector().Add(*labelFilter),
+		}); err != nil {
```

Comment:
We can use `ListResourceSlicesByLabel()` from getters pkg/utils/getters and pass the label selector, but at this point I would directly add a utility function `ListResourceSlicesByClusterID()` that use internally `ListResourceSlicesByLabel()`, like we did with `ListVirtualNodesByClusterID()`.